api/packet_injector.go

```diff
@@ -63,9 +68,74 @@ type PacketParamsReq struct {
 	Type       string
 	Payload    string
 	TrackingID string
-	ID         int64
+	ICMPID     int64
 	Count      int64
 	Interval   int64
+	StartTime  time.Time
+}
+
+// StopInjectionReq stop injection API parameters
+type StopInjectionReq struct {
+	UUID string
+	Src  string
+}
+
+func (pirh *PacketInjectorResourceHandler) Name() string {
+	return "injectpacket"
+}
+
+func (pirh *PacketInjectorResourceHandler) New() Resource {
+	id, _ := uuid.NewV4()
+
+	return &PacketParamsReq{
+		UUID: id.String(),
+	}
+}
+
+func (ppr *PacketParamsReq) ID() string {
+	return ppr.UUID
+}
+
+func (ppr *PacketParamsReq) SetID(id string) {
+	ppr.UUID = id
+}
+
+func (pi *PacketInjectorAPI) Create(r Resource) error {
+	ppr := r.(*PacketParamsReq)
+	host, pp, err := pi.requestToParams(ppr)
+	if err != nil {
+		return err
+	}
+
+	trackingID, err := pi.PIClient.InjectPacket(host, pp)
+	if err != nil {
+		return err
+	}
+
+	ppr.TrackingID = trackingID
+	ppr.StartTime = time.Now()
+
+	go pi.expirePI(ppr.UUID, time.Duration(ppr.Count*ppr.Interval)*time.Millisecond)
```

Comment:
I would prefer to use the same mechanism as capture, meaning listening on etcd events. There is a master election mechanism to handle request to agents and here are speaking directly to PI server through the API.